### PR TITLE
calculate scaled sizes slightly better

### PR DESCRIPTION
**Pull request by chrisbank**
so 100x100 image @55% results in 55x55 instead of 56x56

### CODE REVIEW

**Comment by joelrbrandt** [comment]
@chrisbank All right! Is there a QE plan for this? Part of the plan can certainly be to run the generator-assets-automation tests, but heck, I don't know if they even run anymore!

@mcilroyc given you're deep in the generator weeds right now, would reviewing this be quick for you? Otherwise, I can probably find time later this week.

File: lib/mathutil.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+(function () {
+    "use strict";
+    
+    //source of decimalAdjust from
+    //https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil#Decimal_adjustment
```
Comment (joelrbrandt) [inline lib/mathutil.js]: This license attribution needs to be changed slightly to indicate the license. I would suggest something like:

```
// decimalAdjust function from public domain code on mozilla.org
// Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/ceil#Decimal_adjustment
// Copyright: Any copyright is dedicated to the Public Domain. http://creativecommons.org/publicdomain/zero/1.0/
// License information: https://developer.mozilla.org/en-US/docs/MDN/About#Copyrights_and_licenses
```

And then, at the end of the decimalAdjust function, put 

```
// End of decimalAdjust function, end of code released to the public domain
```

File: lib/mathutil.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        }
+        
+        return decimalAdjust("round", value, -(strLen - 2 - decimalSpot));
+    }
+    
+    //does Math.ceil/floor but rounds the least most decimal to try an elimate the most
+    //egregious floating point errors like Math.ciel(100 * .55) === 56 
+    //because 100 * .55 === 55.00000000000001
+    function roundedCeil(value) {
+        return Math.ceil(roundLeastMostDecimal(value));
```
Comment (mcilroyc) [inline lib/mathutil.js]: Given that we're dealing with calculations on pretty straightforward values, it seems like we could just get away with the round-before-ceil method.  That would avoid having to parse the string value, etc.  example: `Math.ceil((100*.55).toFixed(5))` // 55

Maybe I'm missing some edge cases you've already banged your head against.

**Comment by mcilroyc** [comment]
Back to @chrisbank

**Comment by chrisbank** [comment]
@joelrbrandt : Re testing

QE Plans are mostly run automated tests before and after and see if there is any change. Not all the tests pass and then haven't passed for me for at least a year. Irina will also regress all the bugs and do some general test. We'd like to get automated tests running on the extract assets dialog so they can get included in the normal PS automated tests but there is still some work left to get that working.

File: lib/mathutil.js (hunk 1 above)
Comment (chrisbank) [inline lib/mathutil.js]: will do, thanks for the feedback!

File: lib/mathutil.js (hunk 2 above)
Comment (chrisbank) [inline lib/mathutil.js]: Just wasn't sure what precision to cap it at so I went for the largest.

**Comment by chrisbank** [comment]
@joelrbrandt @mcilroyc Updated the code after going through another bug and then looking in the PS C++ code for the image size dialog and seeing that photoshop rounds the pixels when it scales, not rounds up. Can either of you verify this or know whey ceil was originally used here?

**Comment by joelrbrandt** [comment]
@chrisbank if _anyone_ knows why `ceil` was used, that person would be @DennisKehrig !  He hand-tuned this function to work as best as he could make it work given the time pressure he was under to ship v1.

**Comment by redmunds** [comment]
@mcilroyc We're hoping soon to merge the (internal) PR that relies on this change -- what's the status of merging this PR into master?

**Comment by mcilroyc** [comment]
@redmunds this PR is assigned to @chrisbank so I assume he is still testing his newest change.  If it is ready, I'll merge it.

**Comment by chrisbank** [comment]
@mcilroyc this one is ready for now. We'll have some more 1px bug fixes coming, but this one is good to go. Sorry for the delay
